File: src/main/java/com/qwwuyu/server/ctrl/admin/JavaCtrl.java
package com.qwwuyu.server.ctrl.admin;

import com.qwwuyu.server.bean.User;
import com.qwwuyu.server.configs.Constant;
import com.qwwuyu.server.filter.AuthRequired;
import com.qwwuyu.server.utils.J2EEUtil;
import com.qwwuyu.server.utils.JavaRunUtil;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

@Controller
@RequestMapping("/java")
@AuthRequired(permit = Constant.PERMIT_ADMIN)
public class JavaCtrl {
    @RequestMapping("/result")
    public void toJavaResult(HttpServletRequest request, HttpServletResponse response) {
        User user = J2EEUtil.getUser(request);

        String code = request.getParameter("code");
        code = code == null ? "" : code;
        String html = "<!doctype html><html><body><pre>" + JavaRunUtil.run(code) + "</pre></body></html>";

        response.setContentType("text/html;charset=utf-8");
        response.setHeader("Pragma", "No-cache");
        response.setHeader("Cache-Control", "no-cache");
        response.setDateHeader("Expires", 0);
        try {
            response.getWriter().write(html);
        } catch (IOException ignored) {
        }
    }
}

File: app/src/main/kotlin/com/qwwuyu/gs/controller/admin/JavaCtrl.kt
package com.qwwuyu.gs.controller.admin

import com.qwwuyu.gs.configs.Constant
import com.qwwuyu.gs.filter.AuthRequired
import com.qwwuyu.gs.utils.AppUtil
import com.qwwuyu.lib.utils.JavaRunUtil
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@Controller
@RequestMapping("/java")
@AuthRequired(permit = Constant.PERMIT_ADMIN)
class JavaCtrl {
    @RequestMapping("/result")
    fun toJavaResult(request: HttpServletRequest, response: HttpServletResponse) {
        val code = request.getParameter("code") ?: ""
        val file = AppUtil.nextJavaFile()
        val html = "<!doctype html><html><body><pre>" + JavaRunUtil.run(code, file) + "</pre></body></html>"
        response.contentType = "text/html;charset=utf-8"
        response.setHeader("Pragma", "No-cache")
        response.setHeader("Cache-Control", "no-cache")
        response.setDateHeader("Expires", 0)
        response.writer.write(html)
    }
}